platform/mbed_watchdog_mgr.h

```diff
@@ -0,0 +1,92 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_WATCHDOG_MGR_H
+#define MBED_WATCHDOG_MGR_H
+
+#ifdef DEVICE_WATCHDOG
+
+#include "watchdog_api.h"
+#include "mbed_error.h"
+#include "platform/Callback.h"
+#include "platform/mbed_critical.h"
+#include "LowPowerTicker.h"
+#include "Watchdog.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** \addtogroup platform */
+/** A system timer that will reset the system in the case of system failures or
+ *  malfunctions.
+ *
+ * Example:
+ * @code
+ *
+ * mbed_wdog_manager_start();
```

Comment:
whats the purpose of this code snippet ? only start() invoked, no need to kick or anything else? is it worth having?

Reply:
Watchdog has divided into two, HW watchdog & SW watchdog, HW watchdog register for automated periodic call back from lowpowerticker to refresh hw watchdog regathers API used to start the HW watchdog lowpowerticker.